diff --git a/init.rk3288.rc b/init.rk3288.rc
index 21f4aec..475df65 100755
--- a/init.rk3288.rc
+++ b/init.rk3288.rc
@@ -1,12 +1,20 @@
 on early-init
     # raise cpu&ddr fre.
-    chmod  0644 /sys/devices/system/cpu/cpu0/cpufreq/scaling_governor
+    chmod 0644 /sys/devices/system/cpu/cpu0/cpufreq/scaling_governor
     write /sys/devices/system/cpu/cpu0/cpufreq/scaling_governor performance
     write /dev/video_state p
 
 on property:sys.boot_completed=1
     # recover cpu&ddr fre.
-    chmod  0644 /sys/devices/system/cpu/cpu0/cpufreq/scaling_governor
+    chmod 0644 /sys/class/devfreq/ffa30000.gpu/min_freq
+    chmod 0644 /sys/class/devfreq/ffa30000.gpu/max_freq
+    write /sys/class/devfreq/ffa30000.gpu/min_freq 400000000
+    write /sys/class/devfreq/ffa30000.gpu/max_freq 800000000
+    chmod 0644 /sys/devices/system/cpu/cpu0/cpufreq/scaling_min_freq
+    chmod 0644 /sys/devices/system/cpu/cpu0/cpufreq/scaling_max_freq
+    chmod 0644 /sys/devices/system/cpu/cpu0/cpufreq/scaling_governor
+    write /sys/devices/system/cpu/cpu0/cpufreq/scaling_min_freq 1200000
+    write /sys/devices/system/cpu/cpu0/cpufreq/scaling_max_freq 2088000
     write /sys/devices/system/cpu/cpu0/cpufreq/scaling_governor interactive
     write /dev/video_state n
 
-- 
2.37.2

